checks.d/mongo.py

```diff
@@ -101,21 +119,100 @@ class MongoDb(AgentCheck):
         "uptime": GAUGE,
     }
 
-    V2_ONLY_METRICS = {
-        "globalLock.lockTime": GAUGE,
-        "globalLock.ratio": GAUGE,                  # < 2.2
-        "indexCounters.accesses": RATE,
-        "indexCounters.btree.accesses": RATE,       # < 2.4
-        "indexCounters.btree.hits": RATE,           # < 2.4
-        "indexCounters.btree.misses": RATE,         # < 2.4
-        "indexCounters.btree.missRatio": GAUGE,     # < 2.4
-        "indexCounters.hits": RATE,
-        "indexCounters.misses": RATE,
-        "indexCounters.missRatio": GAUGE,
-        "indexCounters.resets": RATE,
+    DURABILITY_METRICS = {
+        """
+        Journaling-related operations and performance report.
+
+        https://docs.mongodb.org/manual/reference/command/serverStatus/#serverStatus.dur
+        """
+        "dur.commits": GAUGE,
+        "dur.commitsInWriteLock": GAUGE,
+        "dur.compression": GAUGE,
+        "dur.earlyCommits": GAUGE,
+        "dur.journaledMB": GAUGE,
+        "dur.timeMs.dt": GAUGE,
+        "dur.timeMs.prepLogBuffer": GAUGE,
+        "dur.timeMs.remapPrivateView": GAUGE,
+        "dur.timeMs.writeToDataFiles": GAUGE,
+        "dur.timeMs.writeToJournal": GAUGE,
+        "dur.writeToDataFilesMB": GAUGE,
+
+        """
+        Required version > 3.0.0.
+        """
+        "dur.timeMs.commits": GAUGE,
+        "dur.timeMs.commitsInWriteLock": GAUGE,
+    }
+
+    COMMANDS_METRICS = {
+        """
+        ServerStatus use of database commands report.
+        Required version > 3.0.0.
+
+        https://docs.mongodb.org/manual/reference/command/serverStatus/#serverStatus.metrics.commands
+        """
+        # >= 3.0
+        "metrics.commands.count.failed": RATE,
+        "metrics.commands.count.total": GAUGE,
+        "metrics.commands.createIndexes.failed": RATE,
+        "metrics.commands.createIndexes.total": GAUGE,
+        "metrics.commands.delete.failed": RATE,
+        "metrics.commands.delete.total": GAUGE,
+        "metrics.commands.eval.failed": RATE,
+        "metrics.commands.eval.total": GAUGE,
+        "metrics.commands.findAndModify.failed": RATE,
+        "metrics.commands.findAndModify.total": GAUGE,
+        "metrics.commands.insert.failed": RATE,
+        "metrics.commands.insert.total": GAUGE,
+        "metrics.commands.update.failed": RATE,
+        "metrics.commands.update.total": GAUGE,
+    }
+
+    LOCKS_METRICS = {
+        """
+        ServerStatus locks report.
+
+        https://docs.mongodb.org/manual/reference/command/serverStatus/#server-status-locks
+        """
+        "locks.Collection.acquireCount.R": RATE,
```

Comment:
The metrics you're reporting throughout don't quite add up with what I see. It could be a difference between MMAP and WiredTiger (I'm running WiredTiger and am guessing you did this with MMAP?)

Ones I see that you're missing:

    "locks.Collection.acquireCount.r",
    "locks.Collection.acquireCount.w",
    "locks.Database.acquireCount.R",
    "locks.Database.acquireCount.W",
    "locks.Database.acquireWaitCount.R",
    "locks.Database.acquireWaitCount.W",
    "locks.Database.timeAcquiringMicros.R",
    "locks.Database.timeAcquiringMicros.W",
    "locks.Global.acquireCount.R",
    "locks.Global.acquireCount.W",
    "locks.Global.acquireWaitCount.R",
    "locks.Global.acquireWaitCount.W",
    "locks.Global.timeAcquiringMicros.R",
    "locks.Global.timeAcquiringMicros.W",

    "network.bytesIn,
    "network.bytesOut",
    "network.numRequests",

    "metrics.commands._isSelf.failed",
    "metrics.commands._isSelf.total",
    "metrics.commands.aggregate.failed",
    "metrics.commands.aggregate.total",
    "metrics.commands.buildInfo.failed",
    "metrics.commands.buildInfo.total",
    "metrics.commands.collStats.failed",
    "metrics.commands.collStats.total",
    "metrics.commands.dbStats.failed",
    "metrics.commands.dbStats.total",
    "metrics.commands.distinct.failed",
    "metrics.commands.distinct.total",
    "metrics.commands.getLastError.failed",
    "metrics.commands.getLastError.total",
    "metrics.commands.getLog.failed",
    "metrics.commands.getLog.total",
    "metrics.commands.getnonce.failed",
    "metrics.commands.getnonce.total",
    "metrics.commands.isMaster.failed",
    "metrics.commands.isMaster.total",
    "metrics.commands.listCollections.failed",
    "metrics.commands.listCollections.total",
    "metrics.commands.listIndexes.failed",
    "metrics.commands.listIndexes.total",
    "metrics.commands.ping.failed",
    "metrics.commands.ping.total",
    "metrics.commands.renameCollection.failed",
    "metrics.commands.renameCollection.total",
    "metrics.commands.replSetGetStatus.failed",
    "metrics.commands.replSetGetStatus.total",
    "metrics.commands.replSetHeartbeat.failed",
    "metrics.commands.replSetHeartbeat.total",
    "metrics.commands.replSetUpdatePosition.failed",
    "metrics.commands.replSetUpdatePosition.total",
    "metrics.commands.saslContinue.failed",
    "metrics.commands.saslContinue.total",
    "metrics.commands.whatsmyuri.failed",
    "metrics.commands.whatsmyuri.total",
    "metrics.operation.writeConflicts",
    "metrics.repl.buffer.count",
    "metrics.repl.buffer.maxSizeBytes",
    "metrics.repl.buffer.sizeBytes",
    "metrics.repl.preload.docs.num",
    "metrics.repl.preload.docs.totalMillis",
    "metrics.repl.storage.freelist.search.bucketExhausted",
    "metrics.repl.storage.freelist.search.requests",
    "metrics.repl.storage.freelist.search.scanned",